### ISSUE TITLE
Minor refactor of noise's implementation

### ISSUE THREAD

**Issue by rymnc** [status-im/nwaku#1098]
## Proposed changes

1. In the noise handshake process, the sender and receiver can only *either* read *or* write the next message. In [noise_handshake_processing.nim](https://github.com/status-im/nwaku/blob/842cfb3476e8867ef44854e5198922388b2161f3/waku/v2/protocol/waku_noise/noise_handshake_processing.nim#L34-L55), there are 2 variables, `reading`, and `writing`. This could be replaced by just one variable (or an enum), `isReading` or `isWriting`, which would avoid the usages of 

    ```nim
    elif writing:
    ```
    in many [places](https://github.com/status-im/nwaku/blob/842cfb3476e8867ef44854e5198922388b2161f3/waku/v2/protocol/waku_noise/noise_handshake_processing.nim#L140)

2. The [message direction](https://github.com/status-im/nwaku/blob/842cfb3476e8867ef44854e5198922388b2161f3/waku/v2/protocol/waku_noise/noise_types.nim#L94-L96) enum could be replaced with either `inbound` or `outbound`, making it more readable

3. Conversion of the following [conditions](https://github.com/status-im/nwaku/blob/842cfb3476e8867ef44854e5198922388b2161f3/waku/v2/protocol/waku_noise/noise_handshake_processing.nim#L389) into a function called `exists()`, which takes in the keypair, and returns if the keypair is equal to the default (for readability purposes)

## Acceptance criteria
- [ ] Reading/Writing state as one variable
- [ ] Converting D_l and D_r into inbound and outbound, respectively
- [ ] Cleaning up default keypair checks

cc: @s1fr0

**Comment by jm-clius** [comment]
Thanks for creating this, @rymnc!

My 2c worth:
1: No strong opinion, but I lean towards supporting @s1fr0's point in sacrificing a little bit of "best coding practice" for "most readable", even if it does introduce a redundant boolean. An enum `{reading, writing}` could do this too, but implies `case` statements everywhere else.
2. No strong opinion, but would support consistency with spec and similar noise implementations. This is a complex module, so we should assume that readers of the module would have looked at the spec as well.

To get to consensus quicker, perhaps the best way forward then is to implement only (3) and add the remaining two suggestions to [this laundry list](https://github.com/status-im/nwaku/issues/688) of possible refactoring tasks to consider in future?

**Comment by rymnc** [comment]
I agree, I'll start working on 3, we can add the others to the laundry list
